pkg/sql/create_index.go

```diff
@@ -194,9 +194,10 @@ func MakeIndexDescriptor(
 	}
 
 	// Ensure that the index name does not exist before trying to create the index.
-	if err := tableDesc.ValidateIndexNameIsUnique(string(n.Name)); err != nil {
-		return nil, err
+	if idx, _ := tableDesc.FindIndexWithName(n.Name.String()); idx != nil {
```

Comment:
any reason not to use `validateConstraintNameIsNotUsed` here?

Reply:
Good question. I suppose it comes down to whether we want constraints and non-unique indexes to share the same namespace or not. Currently, they don't, and so we don't call that function when renaming an index for instance.